### ISSUE TITLE
build.gradle requires a git-cloned repositiory, but this isn't necessary

### ISSUE THREAD

**Issue by yurivict** [broadinstitute/gatk#6395]
https://github.com/broadinstitute/gatk/blob/master/build.gradle#L156

Please make the above requirement optional, or remove it completely. Packages can only be built from tarballs, and this line is in the way.

**Comment by lbergelson** [comment]
@yurivict We can make this change, but where are you getting the tarball from?  The tarballs that github publishes automatically are insufficient to build gatk.  You have to resolve the runtime resources from git-lfs and I'm not sure how you're doing that without using git.  If you're going to prebuild a special tarball on your own, just put a fake git directory with the appropriate tag for the version name.

**Comment by yurivict** [comment]
I get tarballs from github, and then download dependencies and generate the intermediate tarball internally. We never clone git repositories as you might think due to security issues.

**Comment by lbergelson** [comment]
Do you manually download the files in src/main/resources/large from git-lfs, you need those in order to perform the build.  I'm not sure a good way to get those without using git.  You could of course pull them and then tar them though.

**Comment by yurivict** [comment]
gradle builds the dependency cache archive while online, and then builds a package using this archive while offline.

**Comment by lbergelson** [comment]
What I'm saying is that the tarball that github auto-publishes is missing important files that are managed by git-lfs.  So you need to fetch those somehow.  They're not gradle dependencies and gradle doesn't know how to localize them without using git-lfs which relies on the git directory.  If you have a mechanism for localizing those (and for setting the version manually)  then it would be possible get rid of the git repo requirement.  If you don't then there's no point in overriding the check.

**Comment by lbergelson** [comment]
I pushed a branch `lb_make_git_optional` which lets you skip the git check by specifying `-DversionOverride=whateveryourversionnumberis` as part of the build, play with that and tell me if it's working for your use case.

**Comment by yurivict** [comment]
I see, thanks for pointing this out.

We need to run
```
git archive --format tar.gz --prefix gatk-{hash} -o gatk-{hash}.tar.gz {hash}
```
to retrieve and archive the files.

**Comment by lbergelson** [comment]
Cool, if the that branch works for you then I'll open a PR.

**Comment by lbergelson** [comment]
Remember to set the correct version when you build, otherwise people will give us very confusing bug reports :)

**Comment by davidbenjamin** [comment]
@lbergelson It seems like the line in question could just be deleted, but it's beyond me to know whether there are unintended consequences.

**Comment by lbergelson** [comment]
@davidbenjamin The check for a git repo is important for most use cases because we want to set the appropriate version number which is derived from the git repo.  If people build it without a version or with an incorrect version things get very confusing when there are bugs or issues.

**Comment by yurivict** [comment]
```The check for a git repo is important ```

Packages are not built from the git repo. They are built from tarballs that github creates in this case.

**Comment by davidbenjamin** [comment]
Closed by #6450.
